string      state
float64     location_x
float64     location_y
float64     location_z
